### PR TITLE
Update build option for training in java to enable_training_api

### PR DESCRIPTION
**Pull request by askhade**
### Description
Updating the build option for enabling training in java builds from ENABLE_TRAINING -> ENABLE_TRAINING_APIS.
In the native codebase ENABLE_TRAINING is used for enabling full training and ENABLE_TRAINING_APIS is used for creating the lte builds with training apis. Making the change to sync the naming convention across all the language bindings.

It was a bit confusing to see ENABLE_TRAINING when debugging the android build failures for training. Making this change just to improve readability of logs during debugging.

### CODE REVIEW

**Comment by askhade** [comment]
@Craigacp: Can you please review this PR. Thanks!

File: cmake/onnxruntime_java.cmake [hunk 1]
```diff
@@ -217,6 +223,7 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Android")
       -b build-android.gradle -c settings-android.gradle
```
Comment (edgchen1) [inline cmake/onnxruntime_java.cmake:223]: will the updates to build-android.gradle be in a separate PR?

**Review by edgchen1 — DISMISSED** [review]

**Review by baijumeswani — APPROVED** [review]

File: cmake/onnxruntime_java.cmake (hunk 1 above)
Comment (askhade) [reply cmake/onnxruntime_java.cmake:223]: yes, in the packaging pipeline PR

**Review by edgchen1 — APPROVED** [review]

**Review by Craigacp — APPROVED** [review]
LGTM

**Review by snnn — APPROVED** [review]
